### ISSUE TITLE
Can't build and push image to replicate from github actions workflow.

### ISSUE THREAD

**Issue by one1zero1one** [replicate/cog#1723]
I did a `cog init` with `cog version 0.9.9 (built 2024-06-03T21:35:39+0000)`. `cog build` and `cog push` work okay locally (M2 mbp)

Pushed it to a repo, then ran the action workflow with ubuntu (`cog version 0.9.8 (built 2024-05-17T17:09:03Z`), and I get this:

```
#17 importing to docker
#17 loading layer 5d4427064ecc 294.91kB / 29.15MB
#17 loading layer 146826fa3ca0 65.54kB / 3.51MB
#17 loading layer 3069e121245c 131.07kB / 12.87MB
#17 loading layer 238975b59c5f 244B / 244B
#17 loading layer 216328af4e79 65.54kB / 3.41MB
#17 loading layer 84af80179174 32.77kB / 1.56MB
#17 loading layer 162504b06d1e 131.07kB / 12.37MB
#17 loading layer 4dab0643e877 92B / 92B
#17 loading layer cb4cc6583152 32.77kB / 1.36MB
#17 loading layer 5d4427064ecc 294.91kB / 29.15MB 5.3s done
#17 loading layer 146826fa3ca0 65.54kB / 3.51MB 3.9s done
#17 loading layer 3069e121245c 131.07kB / 12.87MB 2.7s done
#17 loading layer 238975b59c5f 244B / 244B 1.9s done
#17 loading layer 216328af4e79 65.54kB / 3.41MB 1.8s done
#17 loading layer 84af80179174 32.77kB / 1.56MB 1.3s done
#17 loading layer 162504b06d1e 131.07kB / 12.37MB 1.2s done
#17 loading layer 4dab0643e877 92B / 92B 0.3s done
#17 loading layer cb4cc6583152 32.77kB / 1.36MB 0.2s done
#17 DONE 5.3s
#16 exporting to docker image format
#16 sending tarball 6.1s done
#16 DONE 8.9s
Validating model schema...
Adding labels to image...
Unable to determine Git tag
#0 building with "builder-b57ac966-71d8-4035-a5c6-bb1f8bca7e53" instance using docker-container driver
#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 103B done
#1 DONE 0.0s
#2 [internal] load metadata for r8.im/xxx/test:latest
#2 ...
#3 [auth] xxx/test:pull token for r8.im
#3 DONE 0.0s
#2 [internal] load metadata for r8.im/xxx/test:latest
#2 ERROR: failed to authorize: failed to fetch oauth token: unexpected status from GET request to https://r8.im/_token?scope=repository%xxx%2Ftest%3Apull&service=us-docker.pkg.dev: 404 Not Found
------
 > [internal] load metadata for r8.im/xxx/test:latest:
------
Dockerfile:1
--------------------
   1 | >>> FROM r8.im/xxx/test
   2 |     COPY .cog/openapi_schema.json .cog
   3 |     
--------------------
ERROR: failed to solve: r8.im/xxx/test: failed to resolve source metadata for r8.im/xxx/test:latest: failed to authorize: failed to fetch oauth token: unexpected status from GET request to https://r8.im/_token?scope=repository%xxx%2Ftest%3Apull&service=us-docker.pkg.dev: 404 Not Found
ⅹ Failed to add labels to image: exit status 1
Error: Process completed with exit code 1.
```

I tried any combination I could think of of version of cog and runner images, same result.  I tried removing 
I feel this is related to [this](https://github.com/replicate/cog/issues/1718) but in my case this is happening only when building on github runner.

I can't figure out why on the remote runner is starting to build again after `Unable to determine Git tag`, while locally it simply ends.

Thanks for any suggestion on how could I get the github action to succeed.

**Comment by one1zero1one** [comment]
Okay so I figured out the issue, the reason why it goes out to pull the image from a registry - instead of using the locally build one - is the [buildx driver](https://docs.docker.com/reference/cli/docker/buildx/create/#driver), which is `docker-container` by default - when using [setup-cog](https://github.com/replicate/setup-cog).

I don't have enough context to understand if the `docker-container` driver is _required_ (i.e. multi-platform features), but [using `docker` driver ](https://github.com/docker/setup-buildx-action?tab=readme-ov-file#inputs) in the buildx setup got me over this error and being able to successfully build and push cogs. 

I spent many frustrating hours to figure this out, so I'll mention _how_ I debugged - maybe it saves someone else time.
I used `mxschmitt/action-tmate@v3` action to ssh into my runner, and then used `cog debug` and `cog build --debug` to see all the steps (thx to cog developers for exposing this). There I have seen that there are couple of steps, and the last one interprets `FROM r8.im/xxx/test`as to literally go and pull the image, instead of using the local image. Using the right buildx driver makes it do the right thing.
